qa/integration-tests/src/test/java/io/camunda/it/backup/BackupRestoreIT.java

```diff
@@ -0,0 +1,257 @@
+import io.camunda.search.connect.configuration.DatabaseType;
+import io.camunda.webapps.backup.BackupStateDto;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration.TestZeebe;
+import io.camunda.zeebe.util.Either;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.stream.Stream;
+import org.apache.http.HttpHost;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.elasticsearch.client.RestClient;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.SelinuxContext;
+import org.testcontainers.shaded.org.awaitility.Awaitility;
+
+@ZeebeIntegration
+public class BackupRestoreIT {
+  private static final String REPOSITORY_NAME = "test-repository";
+  @TempDir public Path repositoryDir;
+  protected CamundaClient camundaClient;
+
+  @TestZeebe(autoStart = false)
+  protected TestStandaloneCamunda testStandaloneCamunda;
+
+  private RestClient restClient;
+  private ElasticsearchClient esClient;
+  private org.opensearch.client.RestClient osRestClient;
+  private OpenSearchClient opensearchClient;
+  //  private GenericContainer<?> storageContainer;
+  private String storagePath;
+  private BackupRestoreTestConfig config;
+
+  private void setupCamunda(final BackupRestoreTestConfig config) throws IOException {
+    //    startStorageContainer(config);
+    testStandaloneCamunda = new TestStandaloneCamunda();
+    this.config = config;
+    config.configure(testStandaloneCamunda, repositoryDir);
+    testStandaloneCamunda.start();
+    testStandaloneCamunda.awaitCompleteTopology();
+    createSearchClient();
+    createRepository(storagePath);
+  }
+
+  @AfterEach
+  public void afterEach() throws IOException {
+    testStandaloneCamunda.stop();
+    if (restClient != null) {
+      restClient.close();
+      esClient = null;
+    }
+    if (osRestClient != null) {
+      osRestClient.close();
+      opensearchClient = null;
+    }
+  }
+
+  public static Stream<BackupRestoreTestConfig> sources() {
+    return Stream.of(DatabaseType.ELASTICSEARCH)
+        .flatMap(
+            dbType ->
+                Stream.of(RepositoryType.values())
+                    .map(
+                        repositoryType ->
+                            new BackupRestoreTestConfig(dbType, repositoryType, "bucket"))
+                    .limit(1));
+  }
+
+  @ParameterizedTest
+  @MethodSource(value = {"sources"})
+  public void shouldBackupAndRestoreToPreviousState(final BackupRestoreTestConfig config)
+      throws IOException, InterruptedException {
+    // given
+    setupCamunda(config);
+    Thread.sleep(10000);
```

Comment:
:question: Why not use Awaitility with a specific condition? We should avoid `Thread.sleep` in test, it leads to flakiness and doesn't mean anything by itself (meaning harder to understand and maintain).